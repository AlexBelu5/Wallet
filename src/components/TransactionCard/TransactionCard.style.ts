import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    marginVertical: 10,
    borderRadius: 5,
    borderWidth: 2,
    borderColor: 'black',
    padding: 5,
  },
  text: {
    marginBottom: 5,
  },
});
