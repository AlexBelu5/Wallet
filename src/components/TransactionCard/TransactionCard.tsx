import React from 'react';
import {Text, View} from 'react-native';
import {StoreTransaction} from '~/store/transactions/types';
import {normalizeAmount} from '~/utils';

import styles from './TransactionCard.style';

export function TransactionCard({value, receiver, sender}: StoreTransaction) {
  return (
    <View style={styles.container}>
      <Text style={styles.text}>Amount: {normalizeAmount(value)}</Text>
      <Text style={styles.text}>Sender: {sender} </Text>
      <Text>Receiver: {receiver}</Text>
    </View>
  );
}
