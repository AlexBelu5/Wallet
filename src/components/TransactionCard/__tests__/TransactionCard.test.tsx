import React from 'react';
import renderer from 'react-test-renderer';

import {TransactionCard} from '../TransactionCard';

jest.mock('@multiversx/sdk-wallet/out', () => {});
jest.mock('@multiversx/sdk-core/out', () => ({
  TokenTransfer: {
    egldFromBigInteger: (amount: string) => ({
      toPrettyString: () => amount,
    }),
  },
}));

it('TransactionCard renders correctly', () => {
  const receiver = 'testReceiver';
  const sender = 'testSender';
  const value = 'testAmount';
  const tree = renderer
    .create(
      <TransactionCard receiver={receiver} sender={sender} value={value} />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
