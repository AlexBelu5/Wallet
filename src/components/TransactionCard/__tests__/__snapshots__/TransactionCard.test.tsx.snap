// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionCard renders correctly 1`] = `
<View
  style={
    {
      "borderColor": "black",
      "borderRadius": 5,
      "borderWidth": 2,
      "marginVertical": 10,
      "padding": 5,
    }
  }
>
  <Text
    style={
      {
        "marginBottom": 5,
      }
    }
  >
    Amount: 
    testAmount
  </Text>
  <Text
    style={
      {
        "marginBottom": 5,
      }
    }
  >
    Sender: 
    testSender
     
  </Text>
  <Text>
    Receiver: 
    testReceiver
  </Text>
</View>
`;
